### PR TITLE
Prepare for release of `v4.1.0`

### PR DESCRIPTION
**Pull request by ashmaroli**
## Summary

V4.1.0 is almost here.

- [x] Couple of `optimization` PRs to be merged..
- [x] Add the list of contributors.
- [x] History document needs some re-organizing..
- [x] Update Release Post date and filename
- [x] Generate updated version of `docs/_docs/history.md`

### CODE REVIEW

**Comment by DirtyF** [comment]
Nice, wonder if we have a final relevant benchmark to share on all the optimization we added since 4.0?

We could also invite people looking to work with taxonomies within a collection to try the new `jekyll-tagories` plugin.

File: docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Hello Jekyllers!
+
+It's time for yet another release that includes enhancements, optimizations and bug-fixes. Highlights of this release
+are:
+
+* Jekyll now supports rendering excerpts for *pages* in addition to documents and posts.
+* One may now use `:slugified_categories` in their permalink configurations to generate a more apt URL (categories are
+downcased and non-alphanumeric characters replaced by dashes) for their for posts and documents.
+* When you build a site with the `--profle` switch, Jekyll will now additionally output a small table showing the amount
```
Comment (Lominean) [inline docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown:17]: Typo

File: docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+It's time for yet another release that includes enhancements, optimizations and bug-fixes. Highlights of this release
+are:
+
+* Jekyll now supports rendering excerpts for *pages* in addition to documents and posts.
+* The `where_exp` filter got enhanced.  Earlier, one could just use either `and` or `or` once per expression. Now, one
+may use those binary operators multiple times in the filter's expression.
+* Jekyll has a new set of filters based on *its flavor* of the `where` and `where_exp` filters. Named `find` and
+`find_exp` filters respectively, they work similar to their ancestors except that they return **the first object** that
+satisfies the given conditions.
```
Comment (DirtyF) [inline docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown:19]: Can we link to examples in the docs for the usage of the filters?

File: docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown [hunk 3]
```diff
@@ -45,6 +45,14 @@ released gem, please remove `|_config\.yml` from the regular expression in the g
 [filters-docs]: {{ 'docs/liquid/filters/' | relative_url }}
 [page-drop-docs]: {{ 'docs/pages/#for-plugin-developers' | relative_url }}
 
+
+### Have questions?
+
+Please reach out on our [community forum](https://talk.jekyllrb.com)
```
Comment (DirtyF) [inline docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown:51]: 👍

File: docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown (hunk 2 above)
Comment (ashmaroli) [reply docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown:19]: Filters are documented in a single table. Individual filters don't have an anchor to point to.
The link to filter docs page is already linked in the next line (ln#20) of this post

File: docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown (hunk 3 above)
Comment (ashmaroli) [reply docs/_posts/2020-05-25-jekyll-4-1-0-released.markdown:51]: Found this in the Releast Post for `v4.0.0` :wink:

**Pull request merged**